### PR TITLE
Open log view from notifications

### LINKED ISSUE #2643: Error notification hides log panel (and actual error message)
__Describe the Bug__

When I import a broken diagram I did previously get details from the log immediately. With the new placement of notifications major parts of the log contents are hidden. I have to close the notification (not that simple) before accessing it:

capture bZ2m5R_optimized

__Steps to Reproduce__

* Open a broken diagram, i.e. [error-test.bpmn.txt](https://github.com/camunda/camunda-modeler/files/7740664/error-test.bpmn.txt)
* See notification hiding log

__Expected Behavior__

After an error occurs I get straight forward access on the error.

__Environment__

- OS: Any
- Camunda Modeler Version: `dev`
- Execution Platform: Any
- Installed plug-ins: None

### PR DESCRIPTION
**Pull request by smbea**
Closes #2643

https://user-images.githubusercontent.com/25825387/148395624-f8cc599f-a4e2-4ea7-ae49-143e45980fb7.mov

### CODE REVIEW

File: client/src/app/App.js [hunk 1]
```diff
@@ -1197,12 +1197,27 @@ export class App extends PureComponent {
...
+    if (content && content.type === 'button') {
+      contentWithButton = assign({},content);
+
+      const onClick = {
+        onClick:  ()=> {
+          content.props.onClick(); close();
+        }
+      };
+
+      contentWithButton.props = { ...content.props, ...onClick };
```
Comment (pinussilvestrus) [inline client/src/app/App.js:1212]: Mh this looks like monkey patching to me. Is there a possibility to make this behavior part of the `Notification` component?

**Review by andreasgeier — APPROVED** [review]
✅ LGTM

File: client/src/app/notifications/Notification.js [hunk 2]
```diff
@@ -69,6 +70,29 @@ export default class Notification extends PureComponent {
...
+      close
+    } = this.props;
+
+    // allow button to close notification
+    if (content && content.type === 'button') {
+      const contentWithButton = assign({},content);
+
+      const onClick = {
+        onClick:  ()=> {
+          content.props.onClick(); close();
```
Comment (pinussilvestrus) [inline client/src/app/notifications/Notification.js:85]: This still looks like we are monkey patching here. I understand why we are doing it (lazy adding props), but maybe we can rely on React API instead, something like

```js
 const onClick = ()=> {
  content.props.onClick(); close();
};

return React.cloneElement(
  content,
  { onClick }
);
```

File: client/src/plugins/zeebe-plugin/deployment-plugin/__tests__/DeploymentPluginSpec.js [hunk 3]
```diff
@@ -425,7 +425,6 @@ describe('<DeploymentPlugin> (Zeebe)', () => {
     expect(displayNotification).to.have.been.calledOnce;
 
     const notification = displayNotification.getCall(0).args[0];
-    console.log(notification);
```
Comment (pinussilvestrus) [inline client/src/plugins/zeebe-plugin/deployment-plugin/__tests__/DeploymentPluginSpec.js:428]: Nice 😄

File: client/src/app/notifications/Notification.js [hunk 4]
```diff
@@ -69,6 +70,29 @@ export default class Notification extends PureComponent {
...
 
+  getContent() {
+    const {
+      content,
+      close
+    } = this.props;
+
+    // allow button to close notification
+    if (content && content.type === 'button') {
+      const contentWithButton = assign({},content);
```
Comment (pinussilvestrus) [inline client/src/app/notifications/Notification.js:81]: Can we add some test coverage for this in `NotificationSpec`?

File: client/src/app/notifications/Notification.less [hunk 5]
```diff
@@ -51,6 +51,16 @@
     color: var(--color-grey-225-10-35);
     margin-left: 36px;
     margin-right: 16px;
+
+    button {
+      border: none;
+      background-color: transparent;
+      margin: 0;
+      padding: 0;
+      color: var(--color-blue-205-100-50);
```
Comment (pinussilvestrus) [inline client/src/app/notifications/Notification.less:60]: Maybe not only related to this PR, but can we introduce color variables for the Notification styles? Like we do in [other places](https://github.com/camunda/camunda-modeler/blob/develop/client/src/shared/ui/overlay/OverlayDropdown.less#L1). We have a good chance to clean things up here 👍

File: client/src/plugins/zeebe-plugin/deployment-plugin/__tests__/DeploymentPluginSpec.js [hunk 6]
```diff
@@ -459,11 +458,53 @@ describe('<DeploymentPlugin> (Zeebe)', () => {
...
+    // then
+    expect(displayNotificationSpy).to.have.been.calledOnce;
+
+    const notification = displayNotificationSpy.getCall(0).args[0];
+
+    notification.content.props.onClick();
+
+    expect(logSpy).to.have.been.calledOnceWith({
+      category: 'deploy-error',
+      message: mockResult.response.details
```
Comment (pinussilvestrus) [inline client/src/plugins/zeebe-plugin/deployment-plugin/__tests__/DeploymentPluginSpec.js:507]:
```suggestion
    await instance.deploy();

    // assume
    expect(displayNotificationSpy).to.have.been.calledOnce;

    const notification = displayNotificationSpy.getCall(0).args[0];

    // when
    notification.content.props.onClick();

    // then
    expect(logSpy).to.have.been.calledOnceWith({
      category: 'deploy-error',
      message: mockResult.response.details
```

Comment (pinussilvestrus) [reply client/src/plugins/zeebe-plugin/deployment-plugin/__tests__/DeploymentPluginSpec.js:507]: As a proposal, to indicate the test structure

**Review by pinussilvestrus — COMMENTED** [review]
Works well, in general, good job 👍  Some things to consider.

File: client/src/app/notifications/Notification.less (hunk 5 above)
Comment (smbea) [reply client/src/app/notifications/Notification.less]: It's a quick thing that is easily handled now 👍

File: client/src/app/notifications/Notification.js (hunk 4 above)
Comment (smbea) [reply client/src/app/notifications/Notification.js]: Added test case for this

File: client/src/app/notifications/Notification.js (hunk 2 above)
Comment (smbea) [reply client/src/app/notifications/Notification.js]: Smart thinking, makes much more sense. Thanks!

File: client/src/app/notifications/Notification.js [hunk 7]
```diff
@@ -69,6 +69,29 @@ export default class Notification extends PureComponent {
...
 
+  getContent() {
+    const {
+      content,
+      close
+    } = this.props;
+
+    // allow button to close notification
+    if (content && content.type === 'button') {
+      console.log('hey');
```
Comment (pinussilvestrus) [inline client/src/app/notifications/Notification.js:80]: Last thing to be removed, then we're fine 🙂

File: client/src/app/notifications/Notification.less (hunk 5 above)
Comment (pinussilvestrus) [reply client/src/app/notifications/Notification.less]: Thanks a lot 👍

**Review by pinussilvestrus — APPROVED** [review]

**Pull request merged**